src/model/guild/guild_id.rs

```diff
@@ -214,35 +246,18 @@ impl GuildId {
     ///
     /// In addition to the reasons [`Self::ban`] may return an error, may
     /// also return [`Error::ExceededLimit`] if `reason` is too long.
-    #[inline]
+    #[cache_check = if let Some(cache) = cache_http.cache() {
+        self.require_perms(&cache_http, Permissions::BAN_MEMBERS).await?;
+        self.check_hierarchy(cache, user)?;
+    }]
     pub async fn ban_with_reason(
         self,
-        http: impl AsRef<Http>,
-        user: impl Into<UserId>,
-        dmd: u8,
-        reason: impl AsRef<str>,
-    ) -> Result<()> {
-        self._ban(http, user.into(), dmd, Some(reason.as_ref())).await
-    }
-
-    async fn _ban(
-        self,
-        http: impl AsRef<Http>,
+        cache_http: impl CacheHttp,
         user: UserId,
```

Comment:
Same as above.

Reply:
`_ban` is an internal method so there's no concern for API convenience. In fact, `_ban` was using `UserId` directly before this PR already

Reply:
Sorry, the diff there was kinda weird, but this was referring to `ban_with_reason`.